File: src/main/java/club/deneb/client/features/modules/misc/CustomChat.java
package club.deneb.client.features.modules.misc;

import club.deneb.client.Deneb;
import club.deneb.client.event.events.client.PacketEvent;
import club.deneb.client.features.Category;
import club.deneb.client.features.Module;
import club.deneb.client.value.BooleanValue;
import net.minecraft.network.play.client.CPacketChatMessage;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

/**
 * Created by 086 on 8/04/2018.
 */
@Module.Info(name = "CustomChat", category = Category.MISC, description = "Modifies your chat messages")
public class CustomChat extends Module {

    BooleanValue commands = setting("Commands", false);

    @SubscribeEvent
    public void listener (PacketEvent.Send event){
        if (event.getPacket() instanceof CPacketChatMessage) {
            String s = ((CPacketChatMessage) event.getPacket()).getMessage();
            if (s.startsWith("/") && !commands.getValue()) return;
            s += Deneb.CHAT_SUFFIX;
            if (s.length() >= 256) s = s.substring(0,256);
            ((CPacketChatMessage) event.getPacket()).message = s;
        }
    }

}

File: src/main/java/club/deneb/client/features/modules/misc/CustomChat.kt
package club.deneb.client.features.modules.misc

import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraft.network.play.client.CPacketChatMessage
import club.deneb.client.Deneb
import club.deneb.client.event.events.client.PacketEvent
import club.deneb.client.features.Category
import club.deneb.client.features.Module

@Module.Info(name = "CustomChat", category = Category.MISC, description = "Modifies your chat messages")
class CustomChat : Module() {
    private val commands = setting("Commands", false)
    @SubscribeEvent
    fun listener(event: PacketEvent.Send) {
        if (event.packet is CPacketChatMessage) {
            var s = (event.packet as CPacketChatMessage).getMessage()
            if (s.startsWith("/") && !commands.value) return
            s += Deneb.CHAT_SUFFIX
            if (s.length >= 256) s = s.substring(0, 256)
            (event.packet as CPacketChatMessage).message = s
        }
    }
}